First commit Sala

MihaiGhinea opened the pull request.

MihaiGhinea committed:
First commit

GVictorG7 commented on .gitignore:
```diff
@@ -0,0 +1,3 @@
+CODEOWNERS
+target/
+.idea/
```
have a new line at the end of a file

GVictorG7 replied:
new line at the end of a file

GVictorG7 commented on pom.xml:
```diff
@@ -0,0 +1,57 @@
+        </dependency>
+    </dependencies>
+
+</project>
```
have a new line at the end of a file

GVictorG7 commented on pom.xml:
```diff
@@ -0,0 +1,57 @@
+    </parent>
+
+    <groupId>org.example</groupId>
+    <artifactId>Book</artifactId>
```
this app is not about books

GVictorG7 commented on pom.xml:
```diff
@@ -0,0 +1,57 @@
+            <version>3.0.0</version>
+        </dependency>
+        <dependency>
+            <groupId>io.github.cyrilschumacher</groupId>
```
add dependencies from official vendors/providers

GVictorG7 commented on src/main/java/com/hermes/recruitment/config/SwaggerConfig.java:
```diff
@@ -0,0 +1,42 @@
+@Configuration
+public class SwaggerConfig {
+    @Bean
+    public Docket api(TypeResolver typeResolver)
+    {
```
inline brace with method signature

GVictorG7 commented on src/main/java/com/hermes/recruitment/config/SwaggerConfig.java:
```diff
@@ -0,0 +1,42 @@
+                .apiInfo(buildApiInfo());
+    }
+    private ApiInfo buildApiInfo()
+    {
+        return new ApiInfoBuilder()
```
inline brace with method signature

GVictorG7 commented on src/main/java/com/hermes/recruitment/config/SwaggerConfig.java:
```diff
@@ -0,0 +1,42 @@
+                .directModelSubstitute(LocalDateTime.class, java.util.Date.class)
+                .useDefaultResponseMessages(false)
+                .apiInfo(buildApiInfo());
+    }
+    private ApiInfo buildApiInfo()
```
new line between methods

GVictorG7 commented on src/main/java/com/hermes/recruitment/config/SwaggerConfig.java:
```diff
@@ -0,0 +1,42 @@
+import springfox.documentation.spi.DocumentationType;
+import springfox.documentation.spring.web.plugins.Docket;
+
+import java.time.LocalDate;
```
unused import

GVictorG7 commented on src/main/java/com/hermes/recruitment/controller/SalaController.java:
```diff
@@ -0,0 +1,163 @@
+package com.hermes.recruitment.controller;
+
+import com.hermes.recruitment.model.Sala;
+import com.hermes.recruitment.model.dto.SalaDTO;
+import com.hermes.recruitment.service.SalaService;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import lombok.RequiredArgsConstructor;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
```
unused imports

GVictorG7 commented on src/main/java/com/hermes/recruitment/controller/SalaController.java:
```diff
@@ -0,0 +1,163 @@
+            @ApiResponse(code = 204, message = "Nicio sala nu e valabila")}
+    )
+    @GetMapping("/getAll")
+    public ResponseEntity<List<Sala>> getSali()
+    {
```
inline brace with method signature

GVictorG7 commented on src/main/java/com/hermes/recruitment/controller/SalaController.java:
```diff
@@ -0,0 +1,163 @@
+    public ResponseEntity<List<Sala>> getSali()
+    {
+        List<Sala> sali = service.getSali();
+        if (sali.isEmpty())
+            return ResponseEntity.noContent().build();
```
don't omit braces

GVictorG7 commented on src/main/java/com/hermes/recruitment/controller/SalaController.java:
```diff
@@ -0,0 +1,163 @@
+    )
+
+    @GetMapping("/get/{id}")
+    public ResponseEntity<Sala> getSala(@PathVariable Long id)
+    {
```
inline brace with method signature

GVictorG7 commented on src/main/java/com/hermes/recruitment/controller/SalaController.java:
```diff
@@ -0,0 +1,163 @@
+    public ResponseEntity<Sala> getSala(@PathVariable Long id)
+    {
+        Sala sala = service.getSalaId(id);
+        if (sala == null)
+            return ResponseEntity.noContent().build();
```
don't omit braces

GVictorG7 commented on src/main/java/com/hermes/recruitment/controller/SalaController.java:
```diff
@@ -0,0 +1,163 @@
+    )
+
+    @PostMapping("/save")
+    public ResponseEntity<Void> save(@RequestBody List<String> data,  String numar)
```
pass the "Sala" to save by DTO

GVictorG7 commented on src/main/java/com/hermes/recruitment/controller/SalaController.java:
```diff
@@ -0,0 +1,163 @@
+            @ApiResponse(code = 204, message = "Nu a fost gasita")}
+    )
+
+    @PostMapping("/deletesala/{id}")
```
use DELETE method

GVictorG7 commented on src/main/java/com/hermes/recruitment/controller/SalaController.java:
```diff
@@ -0,0 +1,163 @@
+    @PostMapping("/deletesala/{id}")
+    public ResponseEntity<Void> deleteSala(@PathVariable Long id)
+    {
+        Sala book = service.getSalaId(id);
```
don't make 2 separate DB calls
do just the delete and handle the response from there

GVictorG7 commented on src/main/java/com/hermes/recruitment/controller/SalaController.java:
```diff
@@ -0,0 +1,163 @@
+            @ApiResponse(code = 204, message = "Nu a fost gasita")}
+    )
+
+    @PostMapping("/deleteora/{id, ora}")
```
1. wrong mapping
2. wrong path variables definition

GVictorG7 commented on src/main/java/com/hermes/recruitment/model/Sala.java:
```diff
@@ -0,0 +1,38 @@
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import io.swagger.annotations.ApiModelProperty;
+import lombok.*;
```
never import *

GVictorG7 commented on src/main/java/com/hermes/recruitment/model/Sala.java:
```diff
@@ -0,0 +1,38 @@
+    @Column(name = "Id")
+    private long id;
+
+    @Column(name = "Numar", nullable = false, length = 20)
```
db column names are lowercase and with _

GVictorG7 commented on src/main/java/com/hermes/recruitment/model/Sala.java:
```diff
@@ -0,0 +1,38 @@
+    @Id
+    @GeneratedValue
+    @Column(name = "Id")
+    private long id;
```
reference, not primitive

GVictorG7 commented on src/main/java/com/hermes/recruitment/repository/ISalaRepository.java:
```diff
@@ -0,0 +1,12 @@
+
+@Repository
+public interface ISalaRepository extends JpaRepository<Sala, Long> {
+    Sala findByNumar(String numar);
+    Sala getByNumar(String numar);
```
these 2 basically do the same, why have both of them?

GVictorG7 commented on src/main/java/com/hermes/recruitment/service/SalaService.java:
```diff
@@ -0,0 +1,94 @@
+
+    public void save(List<String> data, String numar)
+    {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
```
duplicated literal, extract constant

GVictorG7 commented on src/main/java/com/hermes/recruitment/service/SalaService.java:
```diff
@@ -0,0 +1,94 @@
+        LocalDateTime dateTime = LocalDateTime.parse(ora, formatter);
+        Sala sala = repository.getReferenceById(id);
+        if(repository.findById(id).isPresent())
+            sala.getData().add(dateTime);;
```
trailing semicolon

GVictorG7 commented on src/main/java/com/hermes/recruitment/service/SalaService.java:
```diff
@@ -0,0 +1,94 @@
+        repository.save(sala);
+    }
+
+    public void updateSala(Long id, List<String> data, String numar)
+    {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+        ArrayList<LocalDateTime> date = new ArrayList<>();
+        for (String elem : data)
+        {
+            LocalDateTime dateTime = LocalDateTime.parse(elem, formatter);
+            date.add(dateTime);
+        }
+        deleteSala(id);
+        Sala sala = new Sala(id, numar, date);
+        repository.save(sala);
```
don't need to delete and save for updating
in jpa, saving an entity with an existing ID updates it

GVictorG7 commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,14 @@
+spring.jpa.hibernate.ddl-auto=update
+spring.jpa.show-sql=true
+
+spring.mvc.pathmatch.matching-strategy = ANT_PATH_MATCHER
```
newline at the end of file

GVictorG7 reviewed (changes requested):
for the future, branch names are lowecase

MihaiGhinea committed:
Modify most of the issues in the comments.

Aside from the "save by DTO" one.

Me and Ionut (I think) agreed that it is ok to pass only the list of hours and the number.
Working with a list of LocalDateTime is really tricky and this was the way to bypass it, since the DTO would require said list.

GVictorG7 reviewed (changes requested):
Review the commit message conventions: https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

MihaiGhinea committed:
remove {id, ora}

MihaiGhinea committed:
remove another {id, ora}
